### PR TITLE
Add Storage File Share , File Datalake Perf Tests

### PR DESCRIPTION
**Pull request by g2vinay**
Adds Performane tests for 

1. Storage File Share APIs:
    download
    downloadToFile
    upload
    uploadFromFile

2. Storage File Datalake APis
    read
    append
    upload
    uploadFromFile

### CODE REVIEW

File: sdk/storage/azure-storage-perf/pom.xml [hunk 1]
```diff
@@ -27,8 +27,22 @@
...
       <groupId>com.azure</groupId>
       <artifactId>azure-storage-blob</artifactId>
-      <version>12.10.0-beta.1</version> <!-- {x-version-update;com.azure:azure-storage-blob;current} -->
+      <version>12.9.0</version> <!-- {x-version-update;com.azure:azure-storage-blob;current} -->
     </dependency>
+
+    <dependency>
+      <groupId>com.azure</groupId>
+      <artifactId>azure-storage-file-datalake</artifactId>
+      <version>12.3.0</version> <!-- {x-version-update;com.azure:azure-storage-file-datalake;current} -->
```
Comment (g2vinay) [inline sdk/storage/azure-storage-perf/pom.xml:36]: update to source version

File: sdk/storage/azure-storage-perf/pom.xml [hunk 2]
```diff
@@ -27,8 +27,22 @@
...
+    </dependency>
+
+    <dependency>
+      <groupId>com.azure</groupId>
+      <artifactId>azure-storage-file-share</artifactId>
+      <version>12.7.0</version>
```
Comment (g2vinay) [inline sdk/storage/azure-storage-perf/pom.xml:42]: update to source version.

File: sdk/storage/azure-storage-perf/pom.xml [hunk 3]
```diff
@@ -27,8 +27,22 @@
     <dependency>
       <groupId>com.azure</groupId>
       <artifactId>azure-storage-blob</artifactId>
-      <version>12.10.0-beta.1</version> <!-- {x-version-update;com.azure:azure-storage-blob;current} -->
+      <version>12.9.0</version> <!-- {x-version-update;com.azure:azure-storage-blob;current} -->
```
Comment (g2vinay) [inline sdk/storage/azure-storage-perf/pom.xml:30]: update to source version

Comment (alzimmermsft) [reply sdk/storage/azure-storage-perf/pom.xml:30]: Should we mark this as dependency?

File: sdk/storage/azure-storage-perf/src/main/java/com/azure/storage/file/datalake/perf/AppendFileDatalakeTest.java [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+
+import java.util.UUID;
+
+import static com.azure.perf.test.core.TestDataCreationHelper.createRandomByteBufferFlux;
+
+public class AppendFileDatalakeTest extends DirectoryTest<PerfStressOptions> {
+    private static final int BUFFER_SIZE = 16 * 1024 * 1024;
+    private static final String FILE_NAME = "perfstress-file-" + UUID.randomUUID().toString();
+
+    private final byte[] buffer = new byte[BUFFER_SIZE];
```
Comment (alzimmermsft) [inline sdk/storage/azure-storage-perf/src/main/java/com/azure/storage/file/datalake/perf/AppendFileDatalakeTest.java:21]: This isn't being used

File: sdk/storage/azure-storage-perf/src/main/java/com/azure/storage/file/datalake/perf/ReadFileDatalakeTest.java [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+            .then();
+    }
+
+    // Perform the API call to be tested here
+    @Override
+    public void run() {
+        dataLakeFileClient.read(DEV_NULL);
+    }
+
+    static class NullOutputStream extends OutputStream {
```
Comment (alzimmermsft) [inline sdk/storage/azure-storage-perf/src/main/java/com/azure/storage/file/datalake/perf/ReadFileDatalakeTest.java:49]: Can we move this into the base perf framework, this is likely something we'll want across SDKs.

File: sdk/storage/azure-storage-perf/src/main/java/com/azure/storage/file/datalake/perf/core/FileTestBase.java [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+        dataLakeFileClient =  dataLakeDirectoryClient.getFileClient(fileName);
+        dataLakeFileAsyncClient = dataLakeDirectoryAsyncClient.getFileAsyncClient(fileName);
+    }
+
+    @Override
+    public Mono<Void> setupAsync() {
+        return dataLakeFileAsyncClient.create().then(super.cleanupAsync());
+    }
+
+    public long copyStream(InputStream input, OutputStream out) throws IOException {
```
Comment (alzimmermsft) [inline sdk/storage/azure-storage-perf/src/main/java/com/azure/storage/file/datalake/perf/core/FileTestBase.java:37]: I'm seeing this code in multiple places, can we move it into the base framework as a utility method?

File: sdk/storage/azure-storage-perf/src/main/java/com/azure/storage/file/share/perf/DownloadFileShareTest.java [hunk 7]
```diff
@@ -0,0 +1,84 @@
...
+            .then();
+    }
+
+    // Perform the API call to be tested here
+    @Override
+    public void run() {
+        shareFileClient.download(DEV_NULL);
+    }
+
+    static class NullOutputStream extends OutputStream {
```
Comment (alzimmermsft) [inline sdk/storage/azure-storage-perf/src/main/java/com/azure/storage/file/share/perf/DownloadFileShareTest.java:48]: This is in multiple locations as well

File: sdk/storage/azure-storage-perf/src/main/java/com/azure/storage/file/share/perf/DownloadToFileShareTest.java [hunk 8]
```diff
@@ -0,0 +1,70 @@
...
+import java.net.URISyntaxException;
+import java.util.UUID;
+
+import static com.azure.perf.test.core.TestDataCreationHelper.createRandomByteBufferFlux;
+
+public class DownloadToFileShareTest extends DirectoryTest<PerfStressOptions> {
+    private static final int BUFFER_SIZE = 16 * 1024 * 1024;
+    private static final String FILE_NAME = "perfstress-file-" + UUID.randomUUID().toString();
+
+    private final byte[] buffer = new byte[BUFFER_SIZE];
```
Comment (alzimmermsft) [inline sdk/storage/azure-storage-perf/src/main/java/com/azure/storage/file/share/perf/DownloadToFileShareTest.java:26]: This isn't used

File: sdk/storage/azure-storage-perf/src/main/java/com/azure/storage/file/share/perf/core/FileTestBase.java [hunk 9]
```diff
@@ -0,0 +1,51 @@
...
+
+        String fileName = "randomfiletest-" + UUID.randomUUID().toString();
+
+        shareFileClient =  shareDirectoryClient.getFileClient(fileName);
+        shareFileAsyncClient = shareDirectoryAsyncClient.getFileClient(fileName);
+    }
+
+    @Override
+    public Mono<Void> setupAsync() {
+        return shareFileAsyncClient.create(options.getSize()+DEFAULT_BUFFER_SIZE).then(super.cleanupAsync());
```
Comment (alzimmermsft) [inline sdk/storage/azure-storage-perf/src/main/java/com/azure/storage/file/share/perf/core/FileTestBase.java:33]: Any reason we are adding to the size of the file?

File: sdk/storage/azure-storage-perf/src/main/java/com/azure/storage/file/share/perf/core/ServiceTest.java [hunk 10]
```diff
@@ -0,0 +1,33 @@
...
+
+public abstract class ServiceTest<TOptions extends PerfStressOptions> extends PerfStressTest<TOptions> {
+
+    protected final ShareServiceClient shareServiceClient;
+    protected final ShareServiceAsyncClient shareServiceAsyncClient;
+
+
+    public ServiceTest(TOptions options) {
+        super(options);
+        String connectionString = System.getenv("STORAGE_CONNECTION_STRING");
```
Comment (alzimmermsft) [inline sdk/storage/azure-storage-perf/src/main/java/com/azure/storage/file/share/perf/core/ServiceTest.java:21]: We should use `Configuration` so we can pass `STORAGE_CONNECTION_STRING` via system properties or environment variables

File: sdk/storage/microsoft-azure-storage-perf/src/main/java/com/microsoft/azure/storage/file/share/perf/DownloadFileShareTest.java [hunk 11]
```diff
@@ -0,0 +1,92 @@
...
+
+public class DownloadFileShareTest extends DirectoryTest<PerfStressOptions> {
+    private static final int BUFFER_SIZE = 16 * 1024 * 1024;
+    private static final OutputStream DEV_NULL = new NullOutputStream();
+    private static final String FILE_NAME = "perfstress-file-" + UUID.randomUUID().toString();
+
+    private final CloudFile cloudFile;
+
+
+    private final byte[] buffer = new byte[BUFFER_SIZE];
```
Comment (alzimmermsft) [inline sdk/storage/microsoft-azure-storage-perf/src/main/java/com/microsoft/azure/storage/file/share/perf/DownloadFileShareTest.java:27]: Not used

File: sdk/storage/microsoft-azure-storage-perf/src/main/java/com/microsoft/azure/storage/file/share/perf/DownloadFileShareTest.java [hunk 12]
```diff
@@ -0,0 +1,92 @@
...
+            throw new RuntimeException(e);
+        }
+    }
+
+    // Required resource setup goes here, upload the file to be downloaded during tests.
+
+    @Override
+    public Mono<Void> globalSetupAsync() {
+        return super.globalSetupAsync()
+            .doOnTerminate(() -> {
```
Comment (alzimmermsft) [inline sdk/storage/microsoft-azure-storage-perf/src/main/java/com/microsoft/azure/storage/file/share/perf/DownloadFileShareTest.java:43]: Why are we doing this on terminate?

File: sdk/storage/azure-storage-perf/src/main/java/com/azure/storage/App.java [hunk 13]
```diff
@@ -0,0 +1,47 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.storage;
+
+import com.azure.perf.test.core.PerfStressProgram;
+import com.azure.storage.blob.perf.*;
```
Comment (gapra-msft) [inline sdk/storage/azure-storage-perf/src/main/java/com/azure/storage/App.java:7]: Should we undo this * import? Not sure if this matters cause its just the perf package,

File: sdk/storage/azure-storage-perf/src/main/java/com/azure/storage/file/datalake/perf/AppendFileDatalakeTest.java (hunk 4 above)
Comment (gapra-msft) [reply sdk/storage/azure-storage-perf/src/main/java/com/azure/storage/file/datalake/perf/AppendFileDatalakeTest.java:21]: This and BUFFER_SIZE as a result

File: sdk/storage/azure-storage-perf/src/main/java/com/azure/storage/file/datalake/perf/AppendFileDatalakeTest.java [hunk 14]
```diff
@@ -0,0 +1,57 @@
...
+        super(options);
+        dataLakeFileClient = dataLakeDirectoryClient.getFileClient(FILE_NAME);
+        dataLakeFileAsyncClient = dataLakeDirectoryAsyncClient.getFileAsyncClient(FILE_NAME);
+    }
+
+    // Required resource setup goes here, upload the file to be downloaded during tests.
+    public Mono<Void> globalSetupAsync() {
+        return super.globalSetupAsync()
+            .then(dataLakeFileAsyncClient.create())
+            .then(dataLakeFileAsyncClient.upload(createRandomByteBufferFlux(options.getSize()), null, true))
```
Comment (gapra-msft) [inline sdk/storage/azure-storage-perf/src/main/java/com/azure/storage/file/datalake/perf/AppendFileDatalakeTest.java:36]: Do we have to upload as part of setup?

File: sdk/storage/azure-storage-perf/src/main/java/com/azure/storage/file/datalake/perf/core/FileSystemTest.java [hunk 15]
```diff
@@ -0,0 +1,39 @@
...
+import com.azure.storage.file.datalake.DataLakeFileSystemAsyncClient;
+import com.azure.storage.file.datalake.DataLakeFileSystemClient;
+import com.azure.storage.file.share.ShareAsyncClient;
+import com.azure.storage.file.share.ShareClient;
+import reactor.core.publisher.Mono;
+
+import java.util.UUID;
+
+public abstract class FileSystemTest<TOptions extends PerfStressOptions> extends ServiceTest<TOptions> {
+    private static final String SHARE_NAME = "perfstress-dl-" + UUID.randomUUID().toString();
```
Comment (gapra-msft) [inline sdk/storage/azure-storage-perf/src/main/java/com/azure/storage/file/datalake/perf/core/FileSystemTest.java:16]: FILE_SYSTEM_NAME

File: sdk/storage/microsoft-azure-storage-perf/src/main/java/com/microsoft/azure/storage/blob/perf/DownloadBlobTest.java [hunk 16]
```diff
@@ -0,0 +1,56 @@
...
+                throw new RuntimeException(e);
+            }
+            return 1;
+        })).then();
+    }
+
+    @Override
+    public void run() {
+        try {
+            cloudBlockBlob.download(new NullOutputStream());
```
Comment (mikeharder) [inline sdk/storage/microsoft-azure-storage-perf/src/main/java/com/microsoft/azure/storage/blob/perf/DownloadBlobTest.java:46]: Re-use NullOutputStream

File: sdk/storage/microsoft-azure-storage-perf/src/main/java/com/microsoft/azure/storage/blob/perf/DownloadBlobTest.java [hunk 17]
```diff
@@ -0,0 +1,56 @@
...
+import com.azure.perf.test.core.NullOutputStream;
+import com.azure.perf.test.core.PerfStressOptions;
+import com.azure.perf.test.core.TestDataCreationHelper;
+import com.microsoft.azure.storage.StorageException;
+import com.microsoft.azure.storage.blob.CloudBlockBlob;
+
+import com.microsoft.azure.storage.blob.perf.core.ContainerTest;
+import reactor.core.publisher.Mono;
+
+public class DownloadBlobTest extends ContainerTest<PerfStressOptions> {
```
Comment (mikeharder) [inline sdk/storage/microsoft-azure-storage-perf/src/main/java/com/microsoft/azure/storage/blob/perf/DownloadBlobTest.java:18]: Is this a Track1 test?  If so, I recommend putting Track1 and Track2 tests in completely different projects, and using only Track1 APIs including in setup/cleanup.

File: sdk/storage/microsoft-azure-storage-perf/src/main/java/com/microsoft/azure/storage/blob/perf/UploadFromFileTest.java [hunk 18]
```diff
@@ -0,0 +1,83 @@
...
+            return Mono.empty();
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    @Override
+    public void run() {
+        try {
+            cloudBlockBlob.uploadFromFile(tempFile.toString());
```
Comment (mikeharder) [inline sdk/storage/microsoft-azure-storage-perf/src/main/java/com/microsoft/azure/storage/blob/perf/UploadFromFileTest.java:73]: Cache temp file string.

File: sdk/storage/microsoft-azure-storage-perf/src/main/java/com/microsoft/azure/storage/blob/perf/UploadOutputStreamTest.java [hunk 19]
```diff
@@ -0,0 +1,40 @@
...
+public class UploadOutputStreamTest extends BlobTestBase<PerfStressOptions> {
+
+    public UploadOutputStreamTest(PerfStressOptions options) {
+        super(options);
+    }
+
+    @Override
+    public void run() {
+        try {
+            InputStream inputStream = TestDataCreationHelper.createRandomInputStream(options.getSize());
```
Comment (mikeharder) [inline sdk/storage/microsoft-azure-storage-perf/src/main/java/com/microsoft/azure/storage/blob/perf/UploadOutputStreamTest.java:26]: Cache input stream

File: sdk/storage/microsoft-azure-storage-perf/src/main/java/com/microsoft/azure/storage/file/share/perf/DownloadFileShareTest.java [hunk 20]
```diff
@@ -0,0 +1,88 @@
...
+    }
+
+    @Override
+    public Mono<Void> runAsync() {
+        throw new UnsupportedOperationException();
+    }
+
+    // NOTE: the pattern, cleanup yourself, then the parent.
+    @Override
+    public Mono<Void> globalCleanupAsync() {
```
Comment (mikeharder) [inline sdk/storage/microsoft-azure-storage-perf/src/main/java/com/microsoft/azure/storage/file/share/perf/DownloadFileShareTest.java:82]: Unnecessary

File: sdk/storage/microsoft-azure-storage-perf/src/main/java/com/microsoft/azure/storage/file/share/perf/DownloadToFileShareTest.java [hunk 21]
```diff
@@ -0,0 +1,71 @@
...
+        }
+    }
+
+    @Override
+    public Mono<Void> runAsync() {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public Mono<Void> globalCleanupAsync() {
```
Comment (mikeharder) [inline sdk/storage/microsoft-azure-storage-perf/src/main/java/com/microsoft/azure/storage/file/share/perf/DownloadToFileShareTest.java:65]: Unnecessary

File: sdk/storage/microsoft-azure-storage-perf/src/main/java/com/microsoft/azure/storage/file/share/perf/DownloadToFileShareTest.java [hunk 22]
```diff
@@ -0,0 +1,71 @@
...
+                    throw new RuntimeException(e);
+                }
+            })).then();
+    }
+
+    // Perform the API call to be tested here
+    @Override
+    public void run() {
+        try {
+            File tempFile = File.createTempFile("tempFile", "fileshare");
```
Comment (mikeharder) [inline sdk/storage/microsoft-azure-storage-perf/src/main/java/com/microsoft/azure/storage/file/share/perf/DownloadToFileShareTest.java:51]: Re-use temp file

File: sdk/storage/microsoft-azure-storage-perf/src/main/java/com/microsoft/azure/storage/file/share/perf/UploadFileShareTest.java [hunk 23]
```diff
@@ -0,0 +1,36 @@
...
+
+public class UploadFileShareTest extends FileTestBase<PerfStressOptions> {
+    public UploadFileShareTest(PerfStressOptions options) {
+        super(options);
+    }
+
+    @Override
+    public void run() {
+        try {
+            cloudFile.upload(TestDataCreationHelper.createRandomInputStream(options.getSize()),
```
Comment (mikeharder) [inline sdk/storage/microsoft-azure-storage-perf/src/main/java/com/microsoft/azure/storage/file/share/perf/UploadFileShareTest.java:23]: Cache input stream

File: sdk/storage/microsoft-azure-storage-perf/src/main/java/com/microsoft/azure/storage/file/share/perf/UploadFileShareTest.java [hunk 24]
```diff
@@ -0,0 +1,36 @@
...
+        } catch (StorageException | URISyntaxException | IOException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    @Override
+    public Mono<Void> runAsync() {
+        throw new UnsupportedOperationException();
+    }
+
```
Comment (mikeharder) [inline sdk/storage/microsoft-azure-storage-perf/src/main/java/com/microsoft/azure/storage/file/share/perf/UploadFileShareTest.java:34]: NIT: Extra blank line

File: sdk/storage/microsoft-azure-storage-perf/src/main/java/com/microsoft/azure/storage/file/share/perf/UploadFromFileShareTest.java [hunk 25]
```diff
@@ -0,0 +1,81 @@
...
+    }
+
+    public UploadFromFileShareTest(PerfStressOptions options) {
+        super(options);
+    }
+
+    @Override
+    public void run() {
+        try {
+            cloudFile.uploadFromFile(TEMP_FILE.toString());
```
Comment (mikeharder) [inline sdk/storage/microsoft-azure-storage-perf/src/main/java/com/microsoft/azure/storage/file/share/perf/UploadFromFileShareTest.java:40]: Cache file name string

File: sdk/storage/microsoft-azure-storage-perf/src/main/java/com/microsoft/azure/storage/file/share/perf/core/DirectoryTest.java [hunk 26]
```diff
@@ -0,0 +1,47 @@
...
+        return super.globalSetupAsync()
+            .then(Mono.fromCallable(() -> {
+                cloudFileDirectory.create();
+                return 1; }))
+            .then();
+    }
+
+    // NOTE: the pattern, cleanup yourself, then the parent.
+    @Override
+    public Mono<Void> globalCleanupAsync() {
```
Comment (mikeharder) [inline sdk/storage/microsoft-azure-storage-perf/src/main/java/com/microsoft/azure/storage/file/share/perf/core/DirectoryTest.java:41]: Unnecessary

File: sdk/storage/microsoft-azure-storage-perf/src/main/java/com/microsoft/azure/storage/file/share/perf/core/FileTestBase.java [hunk 27]
```diff
@@ -0,0 +1,56 @@
...
+        int read;
+        while ((read = input.read(buffer, 0, DEFAULT_BUFFER_SIZE)) >= 0) {
+            out.write(buffer, 0, read);
+            transferred += read;
+        }
+        return transferred;
+    }
+
+    @Override
+    public Mono<Void> cleanupAsync() {
```
Comment (mikeharder) [inline sdk/storage/microsoft-azure-storage-perf/src/main/java/com/microsoft/azure/storage/file/share/perf/core/FileTestBase.java:46]: Unnecessary

**Review by mikeharder — CHANGES_REQUESTED** [review]

File: sdk/storage/azure-storage-perf/src/main/java/com/azure/storage/file/datalake/perf/AppendFileDatalakeTest.java [hunk 28]
```diff
@@ -0,0 +1,54 @@
...
+    // Perform the API call to be tested here
+    @Override
+    public void run() {
+        dataLakeFileClient.append(TestDataCreationHelper.createRandomInputStream(options.getSize()),
+            0, options.getSize());
+    }
+
+    @Override
+    public Mono<Void> runAsync() {
+        return dataLakeFileAsyncClient.append(createRandomByteBufferFlux(options.getSize()), 0, options.getSize());
```
[the remark this answers is not in the document]
Comment (alzimmermsft) [reply sdk/storage/azure-storage-perf/src/main/java/com/azure/storage/file/datalake/perf/AppendFileDatalakeTest.java:45]: `randomByteBufferFlux` is safe to re-use as it'll create window `ByteBuffer`s each time it is consumed.

[the remark this answers is not in the document]
Comment (mikeharder) [reply sdk/storage/azure-storage-perf/src/main/java/com/azure/storage/file/datalake/perf/AppendFileDatalakeTest.java:45]: @g2vinay: Let's try reusing the `randomByteBufferFlux`.

File: sdk/storage/azure-storage-perf/src/main/java/com/azure/storage/file/datalake/perf/core/ServiceTest.java [hunk 29]
```diff
@@ -0,0 +1,46 @@
...
+
+public abstract class ServiceTest<TOptions extends PerfStressOptions> extends PerfStressTest<TOptions> {
+
+    protected final DataLakeServiceClient dataLakeServiceClient;
+    protected final DataLakeServiceAsyncClient dataLakeServiceAsyncClient;
+
+
+    public ServiceTest(TOptions options) {
+        super(options);
+        String connectionString = System.getenv("STORAGE_CONNECTION_STRING");
```
[the remark this answers is not in the document]
Comment (g2vinay) [reply sdk/storage/azure-storage-perf/src/main/java/com/azure/storage/file/datalake/perf/core/ServiceTest.java]: We already have it in azure-core, updated to use that.

File: sdk/storage/azure-storage-perf/src/main/java/com/azure/storage/file/share/perf/DownloadToFileShareTest.java [hunk 30]
```diff
@@ -0,0 +1,58 @@
...
+        return super.globalSetupAsync()
+            .then(shareFileAsyncClient.create(options.getSize()))
+            .then(shareFileAsyncClient.upload(createRandomByteBufferFlux(options.getSize()), options.getSize()))
+            .then();
+    }
+
+    // Perform the API call to be tested here
+    @Override
+    public void run() {
+        File file = new File(UUID.randomUUID().toString());
```
[the remark this answers is not in the document]
Comment (g2vinay) [reply sdk/storage/azure-storage-perf/src/main/java/com/azure/storage/file/share/perf/DownloadToFileShareTest.java:40]: The API expects the file to not exist already, so we can't use a single file per thread here.
The Track 1 API can reuse the same file, so I'll file an issue here for the storage team to look at.

File: sdk/storage/azure-storage-perf/src/main/java/com/azure/storage/file/datalake/perf/AppendFileDatalakeTest.java (hunk 28 above)
[the remark this answers is not in the document]
Comment (g2vinay) [reply sdk/storage/azure-storage-perf/src/main/java/com/azure/storage/file/datalake/perf/AppendFileDatalakeTest.java]: For datalake API, reusing the byte buffer flux, doesn't work, for other APIs, it works.
I'll file an issue for this.

**Review by mikeharder — CHANGES_REQUESTED** [review]

**Review by mikeharder — APPROVED** [review]

File: sdk/storage/azure-storage-perf/src/main/java/com/azure/storage/file/share/perf/core/ShareTest.java [hunk 31]
```diff
@@ -0,0 +1,33 @@
...
+        super(options);
+        // Setup the container clients
+        String shareName = "perfstress-sharev11-" + UUID.randomUUID().toString();
+        shareClient = shareServiceClient.getShareClient(shareName);
+        shareAsyncClient = shareServiceAsyncClient.getShareAsyncClient(shareName);
+    }
+
+    @Override
+    public Mono<Void> setupAsync() {
+        return super.setupAsync().then(Mono.defer(() -> {
```
Comment (mikeharder) [inline sdk/storage/azure-storage-perf/src/main/java/com/azure/storage/file/share/perf/core/ShareTest.java:29]: Can this use the same (and simpler) pattern as DataLake?

https://github.com/Azure/azure-sdk-for-java/pull/17883/files#diff-10b82805ef0ae91ec67e7271940bf4767256bb9e37a9833e7f27d343332eb40aR29

**Review by mikeharder — CHANGES_REQUESTED** [review]

**Review by mikeharder — APPROVED** [review]

**Pull request merged**